elastic/kibana
# [Alerting] Adding a configurable minimal rule interval for newly created rules

Resolves https://github.com/elastic/kibana/issues/124810

## Summary

* Removed `minimumScheduleInterval` from rule type registration since it was not used by any rule type
* Added `minimumScheduleInterval` to `kibana.yml` config
* Updated server side validation to use configured value instead of rule type value
* Updated client side validation to use configured value instead of rule type value
  * Added new route for getting this config inside `triggers_actions_ui` plugin

## To verify:
* Verify that rules created before this PR with an interval less than the minimum continue running
* Verify that when updating existing rules with too-short intervals, the minimum is enforced.
* Verify that you cannot create a rule via the UI or API with a schedule less than the minimum.

## Breaking Change
We previously did not enforce any minimum schedule interval for rules. Adding this minimum may be a breaking change for users using automated methods to create or update their rules. If users are impacted by this change in behavior, they have the option of configuring `xpack.alerting.minimumScheduleInterval` to be a smaller value or increasing the interval of their alerting rules. 

### Checklist
